### ISSUE TITLE
Async (TS) samples are not working with Jasmine and Expect

### ISSUE THREAD

**Issue by wswebcreation** [webdriverio/webdriverio#6839]
**Describe the bug**
When using the default examples from WDIO in combination with Jasmine the following code fails

`await expect(SecurePage.flashAlert).toHaveTextContaining('You logged into a secure area!');`

Because the promise is not resolved before executing the assertion

**To Reproduce**
Set up a new project by following the instructions from https://webdriver.io/docs/gettingstarted and choose the following options:

```log
=========================
WDIO Configuration Helper
=========================

? Where is your automation backend located? On my local machine
? Which framework do you want to use? jasmine
? Do you want to use a compiler? TypeScript (https://www.typescriptlang.org/)
? Where are your test specs located? ./test/specs/**/*.ts
? Do you want WebdriverIO to autogenerate some test files? Yes
? Do you want to use page objects (https://martinfowler.com/bliki/PageObject.html)? Yes
? Where are your page objects located? ./test/pageobjects/**/*.ts
? Which reporter do you want to use? spec
? Do you want to add a service to your test setup? chromedriver
? What is the base url? http://localhost

```

It then mentions `npx wdio run wdio.conf.js`, which needs to be `npx wdio run wdio.conf.ts`

**Expected behavior**
The promises for `expect-webdriverio` are resolved before using the matchers

**Log**
If applicable, add logs to help explain your problem.
Create a [gist](https://gist.github.com) which is a paste of your _full_ Selenium/WebdriverIO logs, and link them here. Do _not_ paste your full Appium logs here, as it will make this issue very long and hard to read! If you are reporting a bug, _always_ include Selenium/WebdriverIO logs!

**Additional context**
```log
npx wdio run wdio.conf.ts

Execution of 1 spec files started at 2021-05-10T10:18:14.174Z

2021-05-10T10:18:14.230Z INFO @wdio/cli:launcher: Run onPrepare hook
Starting ChromeDriver 90.0.4430.24 (4c6d850f087da467d926e8eddb76550aed655991-refs/branch-heads/4430@{#429}) on port 9515
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
2021-05-10T10:18:14.351Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2021-05-10T10:18:14.351Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,wdio.conf.ts
[0-0] 2021-05-10T10:18:14.774Z INFO @wdio/local-runner: Run worker command: run
[0-0] RUNNING in chrome - /test/specs/example.e2e.ts
[0-0] 2021-05-10T10:18:15.163Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2021-05-10T10:18:15.163Z INFO webdriver: [POST] http://localhost:9515/session
[0-0] 2021-05-10T10:18:15.163Z INFO webdriver: DATA {
[0-0]   capabilities: {
[0-0]     alwaysMatch: { browserName: 'chrome', acceptInsecureCerts: true },
[0-0]     firstMatch: [ {} ]
[0-0]   },
[0-0]   desiredCapabilities: { browserName: 'chrome', acceptInsecureCerts: true }
[0-0] }
[0-0] 2021-05-10T10:18:17.281Z INFO webdriver: COMMAND navigateTo("https://the-internet.herokuapp.com/login")
[0-0] 2021-05-10T10:18:17.282Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/url
[0-0] 2021-05-10T10:18:17.282Z INFO webdriver: DATA { url: 'https://the-internet.herokuapp.com/login' }
[0-0] 2021-05-10T10:18:20.043Z INFO webdriver: COMMAND findElement("css selector", "#username")
[0-0] 2021-05-10T10:18:20.043Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element
[0-0] 2021-05-10T10:18:20.043Z INFO webdriver: DATA { using: 'css selector', value: '#username' }
[0-0] 2021-05-10T10:18:20.056Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '98425206-3be5-46f7-a7b1-0de781666e0f'
[0-0] }
[0-0] 2021-05-10T10:18:20.061Z INFO webdriver: COMMAND elementClear("98425206-3be5-46f7-a7b1-0de781666e0f")
[0-0] 2021-05-10T10:18:20.061Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element/98425206-3be5-46f7-a7b1-0de781666e0f/clear
[0-0] 2021-05-10T10:18:20.086Z INFO webdriver: COMMAND elementSendKeys("98425206-3be5-46f7-a7b1-0de781666e0f", "<Screenshot[base64]>")
[0-0] 2021-05-10T10:18:20.086Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element/98425206-3be5-46f7-a7b1-0de781666e0f/value
[0-0] 2021-05-10T10:18:20.086Z INFO webdriver: DATA { text: 'tomsmith' }
[0-0] 2021-05-10T10:18:20.224Z INFO webdriver: COMMAND findElement("css selector", "#password")
[0-0] 2021-05-10T10:18:20.224Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element
[0-0] 2021-05-10T10:18:20.224Z INFO webdriver: DATA { using: 'css selector', value: '#password' }
[0-0] 2021-05-10T10:18:20.232Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '01be7ef9-4272-428e-a269-72bd516feb01'
[0-0] }
[0-0] 2021-05-10T10:18:20.234Z INFO webdriver: COMMAND elementClear("01be7ef9-4272-428e-a269-72bd516feb01")
[0-0] 2021-05-10T10:18:20.234Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element/01be7ef9-4272-428e-a269-72bd516feb01/clear
[0-0] 2021-05-10T10:18:20.257Z INFO webdriver: COMMAND elementSendKeys("01be7ef9-4272-428e-a269-72bd516feb01", "SuperSecretPassword!")
[0-0] 2021-05-10T10:18:20.257Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element/01be7ef9-4272-428e-a269-72bd516feb01/value
[0-0] 2021-05-10T10:18:20.257Z INFO webdriver: DATA { text: 'SuperSecretPassword!' }
[0-0] 2021-05-10T10:18:20.308Z INFO webdriver: COMMAND findElement("css selector", "button[type="submit"]")
[0-0] 2021-05-10T10:18:20.308Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element
[0-0] 2021-05-10T10:18:20.308Z INFO webdriver: DATA { using: 'css selector', value: 'button[type="submit"]' }
[0-0] 2021-05-10T10:18:20.316Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '627ab543-42a4-493d-86f1-406db2db7c77'
[0-0] }
[0-0] 2021-05-10T10:18:20.317Z INFO webdriver: COMMAND elementClick("627ab543-42a4-493d-86f1-406db2db7c77")
[0-0] 2021-05-10T10:18:20.317Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element/627ab543-42a4-493d-86f1-406db2db7c77/click
[0-0] 2021-05-10T10:18:20.694Z INFO webdriver: COMMAND findElement("css selector", "#flash")
[0-0] 2021-05-10T10:18:20.695Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element
[0-0] 2021-05-10T10:18:20.695Z INFO webdriver: DATA { using: 'css selector', value: '#flash' }
[0-0] 2021-05-10T10:18:20.696Z INFO webdriver: COMMAND findElement("css selector", "#flash")
[0-0] 2021-05-10T10:18:20.696Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/element
[0-0] 2021-05-10T10:18:20.696Z INFO webdriver: DATA { using: 'css selector', value: '#flash' }
[0-0] Error in "My Login application should login with valid credentials"
Error: Expected [object Promise] to be existing.
    at <Jasmine>
    at <Jasmine>
    at /Users/wimselles/Sauce/Git/appium-boilerplate/.tmp/test/specs/example.e2e.ts:9:45
    at Generator.next (<anonymous>)
[0-0] 2021-05-10T10:18:20.700Z INFO webdriver: COMMAND deleteSession()
[0-0] 2021-05-10T10:18:20.700Z INFO webdriver: [DELETE] http://localhost:9515/session/6b117cd714324517f85788847f258356
[0-0] 2021-05-10T10:18:20.707Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '6639df39-ad33-4fa3-93d2-c943a5900ed1'
[0-0] }
[0-0] 2021-05-10T10:18:20.709Z INFO webdriver: COMMAND getElementText("6639df39-ad33-4fa3-93d2-c943a5900ed1")
[0-0] 2021-05-10T10:18:20.709Z INFO webdriver: [GET] http://localhost:9515/session/6b117cd714324517f85788847f258356/element/6639df39-ad33-4fa3-93d2-c943a5900ed1/text
[0-0] 2021-05-10T10:18:20.711Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '6639df39-ad33-4fa3-93d2-c943a5900ed1'
[0-0] }
[0-0] 2021-05-10T10:18:20.716Z INFO webdriver: COMMAND findElements("css selector", "#flash")
[0-0] 2021-05-10T10:18:20.716Z INFO webdriver: [POST] http://localhost:9515/session/6b117cd714324517f85788847f258356/elements
[0-0] 2021-05-10T10:18:20.716Z INFO webdriver: DATA { using: 'css selector', value: '#flash' }
[0-0] 2021-05-10T10:18:20.791Z ERROR webdriver: Request failed with status 404 due to invalid session id: invalid session id
[0-0] 2021-05-10T10:18:20.793Z ERROR webdriver: Request failed with status 404 due to invalid session id: invalid session id
[0-0] FAILED in chrome - /test/specs/example.e2e.ts
2021-05-10T10:18:20.886Z INFO @wdio/cli:launcher: Run onComplete hook

 "spec" Reporter:
------------------------------------------------------------------
[chrome 90.0.4430.93 mac os x #0-0] Running: chrome (v90.0.4430.93) on mac os x
[chrome 90.0.4430.93 mac os x #0-0] Session ID: 6b117cd714324517f85788847f258356
[chrome 90.0.4430.93 mac os x #0-0]
[chrome 90.0.4430.93 mac os x #0-0] » /test/specs/example.e2e.ts
[chrome 90.0.4430.93 mac os x #0-0] My Login application
[chrome 90.0.4430.93 mac os x #0-0]    ✖ should login with valid credentials
[chrome 90.0.4430.93 mac os x #0-0]
[chrome 90.0.4430.93 mac os x #0-0] 1 failing (3.4s)
[chrome 90.0.4430.93 mac os x #0-0]
[chrome 90.0.4430.93 mac os x #0-0] 1) My Login application should login with valid credentials
[chrome 90.0.4430.93 mac os x #0-0] Expected [object Promise] to be existing.
[chrome 90.0.4430.93 mac os x #0-0] Error: Expected [object Promise] to be existing.
[chrome 90.0.4430.93 mac os x #0-0]     at <Jasmine>
[chrome 90.0.4430.93 mac os x #0-0]     at <Jasmine>
[chrome 90.0.4430.93 mac os x #0-0]     at /Users/wimselles/Sauce/Git/appium-boilerplate/.tmp/test/specs/example.e2e.ts:9:45
[chrome 90.0.4430.93 mac os x #0-0]     at Generator.next (<anonymous>)
[chrome 90.0.4430.93 mac os x #0-0] Expected [object Promise] to have text containing 'You logged into a secure area!'.
[chrome 90.0.4430.93 mac os x #0-0] Error: Expected [object Promise] to have text containing 'You logged into a secure area!'.
[chrome 90.0.4430.93 mac os x #0-0]     at <Jasmine>
[chrome 90.0.4430.93 mac os x #0-0]     at <Jasmine>
[chrome 90.0.4430.93 mac os x #0-0]     at /Users/wimselles/Sauce/Git/appium-boilerplate/.tmp/test/specs/example.e2e.ts:10:45
[chrome 90.0.4430.93 mac os x #0-0]     at Generator.next (<anonymous>)
[chrome 90.0.4430.93 mac os x #0-0] Expected [object Promise] to be existing.
[chrome 90.0.4430.93 mac os x #0-0] error properties: Object({ matcherName: 'toBeExisting', passed: undefined, expected: [  ], actual: [object Promise] })
[chrome 90.0.4430.93 mac os x #0-0] Error: Expected [object Promise] to be existing.
[chrome 90.0.4430.93 mac os x #0-0]     at <Jasmine>
[chrome 90.0.4430.93 mac os x #0-0]     at <Jasmine>
[chrome 90.0.4430.93 mac os x #0-0]     at /Users/wimselles/Sauce/Git/appium-boilerplate/.tmp/test/specs/example.e2e.ts:9:45
[chrome 90.0.4430.93 mac os x #0-0]     at Generator.next (<anonymous>)


Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:06 

2021-05-10T10:18:20.887Z INFO @wdio/local-runner: Shutting down spawned worker
2021-05-10T10:18:21.140Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2021-05-10T10:18:21.140Z INFO @wdio/local-runner: shutting down
```

> Please use the correct markdown

**Comment by chanyk-joseph** [comment]
Hi @christian-bromann, the same issue also happens in (async, js, jasmine, expect) sample
I suspect the root cause is the same

`
 "spec" Reporter:
------------------------------------------------------------------
[chrome 84.0.4147.125 linux #0-0] Running: chrome (v84.0.4147.125) on linux
[chrome 84.0.4147.125 linux #0-0] Session ID: d731f839ad4130c2d45c85d6d35f8e1c
[chrome 84.0.4147.125 linux #0-0]
[chrome 84.0.4147.125 linux #0-0] » /test/specs/example.e2e.js
[chrome 84.0.4147.125 linux #0-0] My Login application
[chrome 84.0.4147.125 linux #0-0]    ✖ should login with valid credentials
[chrome 84.0.4147.125 linux #0-0]
[chrome 84.0.4147.125 linux #0-0] 1 failing (3.5s)
[chrome 84.0.4147.125 linux #0-0]
[chrome 84.0.4147.125 linux #0-0] 1) My Login application should login with valid credentials
[chrome 84.0.4147.125 linux #0-0] Expected [object Promise] to be existing.
[chrome 84.0.4147.125 linux #0-0] Error: Expected [object Promise] to be existing.
[chrome 84.0.4147.125 linux #0-0]     at <Jasmine>
[chrome 84.0.4147.125 linux #0-0]     at <Jasmine>
[chrome 84.0.4147.125 linux #0-0]     at UserContext.<anonymous> (/vagrant/workspace/vos_app_unified_origin_engine-trunk-SanityBuild2/vos_app_unified_origin_engine/tests/basic_ui_test_2/test/specs/example.e2e.js:11:45)
[chrome 84.0.4147.125 linux #0-0]     at processTicksAndRejections (internal/process/task_queues.js:93:5)
[chrome 84.0.4147.125 linux #0-0] Expected [object Promise] to have text containing 'You logged into a secure area!'.
[chrome 84.0.4147.125 linux #0-0] Error: Expected [object Promise] to have text containing 'You logged into a secure area!'.
[chrome 84.0.4147.125 linux #0-0]     at <Jasmine>
[chrome 84.0.4147.125 linux #0-0]     at <Jasmine>
[chrome 84.0.4147.125 linux #0-0]     at UserContext.<anonymous> (/vagrant/workspace/vos_app_unified_origin_engine-trunk-SanityBuild2/vos_app_unified_origin_engine/tests/basic_ui_test_2/test/specs/example.e2e.js:12:45)
[chrome 84.0.4147.125 linux #0-0]     at processTicksAndRejections (internal/process/task_queues.js:93:5)
[chrome 84.0.4147.125 linux #0-0] Expected [object Promise] to be existing.
[chrome 84.0.4147.125 linux #0-0] error properties: Object({ matcherName: 'toBeExisting', passed: undefined, expected: [  ], actual: [object Promise] })
[chrome 84.0.4147.125 linux #0-0] Error: Expected [object Promise] to be existing.
[chrome 84.0.4147.125 linux #0-0]     at <Jasmine>
[chrome 84.0.4147.125 linux #0-0]     at <Jasmine>
[chrome 84.0.4147.125 linux #0-0]     at UserContext.<anonymous> (/vagrant/workspace/vos_app_unified_origin_engine-trunk-SanityBuild2/vos_app_unified_origin_engine/tests/basic_ui_test_2/test/specs/example.e2e.js:11:45)
[chrome 84.0.4147.125 linux #0-0]     at processTicksAndRejections (internal/process/task_queues.js:93:5)


Spec Files:      0 passed, 1 failed, 1 total (100% completed) in 00:00:04 
`

The same issue does not happen when using **mocha** framework
`
 "spec" Reporter:
------------------------------------------------------------------
[chrome 84.0.4147.125 linux #0-0] Running: chrome (v84.0.4147.125) on linux
[chrome 84.0.4147.125 linux #0-0] Session ID: 8532fb500d26d5878e782a05f419246a
[chrome 84.0.4147.125 linux #0-0]
[chrome 84.0.4147.125 linux #0-0] » /test/specs/example.e2e.js
[chrome 84.0.4147.125 linux #0-0] My Login application
[chrome 84.0.4147.125 linux #0-0]    ✓ should login with valid credentials
[chrome 84.0.4147.125 linux #0-0]
[chrome 84.0.4147.125 linux #0-0] 1 passing (3s)


Spec Files:      1 passed, 1 total (100% completed) in 00:00:04
`

**Comment by christian-bromann** [comment]
@mgrybyk any idea? Help on this matter would be appreciated. I assume this issue is related to our `expect-webdriverio` lib.

**Comment by christian-bromann** [comment]
Fixed in #6855
